src/Chainweb/Rosetta/Internal.hs

```diff
@@ -351,66 +383,105 @@ nonGenesisTransaction logs initial rest target
+--     If the TxId of this coin table log matches the remediation's TxId,
+--       then assumes that this log corresponds to the given remediation.
+--     Otherwise, a rosetta transaction is created for the given
+--       remediation with an empty operations list.
+-- NOTE: Assumes that each remediation must have been successful and thus
+--       would have a TxId associated with it.
+-- NOTE: Not all remediations transactions will output coin table logs.
+remediationAcc
+    :: AccumulatorType (TxAccumulator rosettaTx)
+    -> TxAccumulator rosettaTx
+    -> (Command payload, TxId)
+    -> TxAccumulator rosettaTx
+remediationAcc accTyp acc (remTx, remTid) =
+  case _txAccumulator_logsLeft acc of
+    (logTid,logs):rest
+      | logTid == remTid -> -- remediation touched coin table
+        let ops = indexedOperations $!
+              UnindexedOperations
+              { _unindexedOperation_fundOps = []
+              , _unindexedOperation_transferOps = makeOps (logTid, logs)
+              , _unindexedOperation_gasOps = []
+              }
+            rosettaTx = rosettaTransactionFromCmd remTx $! ops
+        in makeAcc rest rosettaTx
+    rest -> -- list of logs empty or remediation didn't touch coin table
+      makeAcc rest $!
+      rosettaTransactionFromCmd remTx []
+  where
+    makeAcc restLogs rosettaTx =
+      accumulatorFunction accTyp acc restLogs rosettaTx
+    makeOps (tid, logs) =
+      map (operation Remediation TransferOrCreateAcct tid) logs
+
+-- | Matches all transactions in a remediation block
+--   (including coinbase, remediations, and user transactions)
+--   to their coin table logs.
+--  Coinbase logs are matched first, followed by remediations txs,
+--  and finally user transactions are matched using the same algorithm
+--  as non-genesis transactions.
+-- NOTE: Assumes that each remediation transaction incremented
+--       the TxId counter by one.
+-- NOTE: Assumes remediations don't occur in blocks where coinbase transaction
+--       could have failed. It needs to know the TxId of the coinbase transaction in
+--       order to derive the remediation's TxIds.
 remediations
     :: Map TxId [AccountLog]
+    -> ChainId
     -> CoinbaseTx (CommandResult Hash)
-    -> Either RosettaFailure [Transaction]
-remediations logs initial =
-    overwriteError RosettaMismatchTxLogs work
+    -> [Command payload]
+    -- ^ Remediation transactions.
+    -- ^ NOTE: No CommandResult available for these.
+    -> V.Vector (CommandResult Hash)
+    -- ^ User transactions in the same block as remediations
+    -> Either String [Transaction]
+remediations logs cid coinbase remTxs txs = do
+  TxAccumulator restLogs coinbaseTx <- nonGenesisCoinbaseLog logsList cid coinbase
+  coinbaseTxId <- note "remediations: No TxId found for Coinbase" (_crTxId coinbase)
+
+  let remWithTxIds = zip remTxs [(succ coinbaseTxId)..]
+      -- ^ Assumes that each remediation transaction gets its own TxId
```

Comment:
And assumes that txids are going to be sequential for each remediation. Is this a correct assumption?

Reply:
Great question! I think I was also asking myself the same thing (see this comment https://github.com/kadena-io/chainweb-node/pull/1135#discussion_r482517990).

I also added this comment to the function description for `remediations`:
> NOTE: Assumes that each remediation transaction incremented the TxId counter by one.

I know that for the remediations that did occur this assumption seemed to hold.

This question might be a good candidate for pair reviewing session though.

Future remediations might not need to use this logic if we realize that Events give us all the information we need.

Reply:
Resolved via: 
> This assumption holds, since the remediation txs are applied directly after coinbase at a particular height, as part of applyCoinbase. We construct the blocks, hence, the txids are not random.

See https://github.com/kadena-io/chainweb-node/blob/c0c300a64040390d603f1183eac126e3bbfebe8d/src/Chainweb/Pact/TransactionExec.hs#L328